source/agora/consensus/EnrollmentManager.d

```diff
@@ -171,6 +184,82 @@ public class EnrollmentManager
         Utils.getCollectorRegistry().addCollector(&this.collectValidatorStats);
     }
 
+    /***************************************************************************
+
+        Update the validator key index maps for the provided height
+
+    ***************************************************************************/
+
+    public void updateValidatorIndexMaps (Height height) @safe
+    {
+        PublicKey[] keys;
+        if (!this.getActiveValidatorPublicKeys(keys, height))
+            assert(0, "Database failure on fetching Validator Public Keys");
+
+        if (keys.length == 0)
+            log.error("No Active validator public keys at height {}", height);
+
+        foreach (idx, key; keys)
+        {
+            log.trace("Update validator lookup maps at height {} for index {} pubkey {}", height, idx, PublicKey(key[]));
+            const K = Point(key[]);
+            this.key_to_index[height][K] = idx;
+            this.index_to_key[height][idx] = K;
+        }
+    }
+
+    /***************************************************************************
+
+        Params:
+            height = the height at which to look up the mapping for
+
+        Returns:
+            the count of validators at this enrollment height
+
+    ***************************************************************************/
+
+    public size_t getCountOfValidators (Height height) nothrow @safe
+    {
+        return height !in this.key_to_index ? 0
+            : this.index_to_key[height].length;
+    }
+
+    /***************************************************************************
+
+        Params:
+            height = the height at which to look up the mapping for
+            K = the Key for which to find the bitfield index to use
+
+        Returns:
+            the index of this key, or ulong.max if none was found
+
+    ***************************************************************************/
+
+    public ulong getIndexOfValidator (Height height, Point K) nothrow @safe
+    {
+        return height !in this.key_to_index
+            || K !in this.key_to_index[height] ?  ulong.max
+            : this.key_to_index[height][K];
```

Comment:
Prefer adding parentheses in a complex one-line like this.

```
return ((height !in this.key_to_index)
            || (K !in this.key_to_index[height])) ?  ulong.max
            : this.key_to_index[height][K];
```

Alternatively since this is already so complex I would prefer if you used `if/else` here. It's very hard to read.

Reply:
Updated to if / else